lib/rules/require-capitalized-constructors.js

```diff
@@ -32,16 +36,19 @@ var assert = require('assert');
 module.exports = function() {};
 
 module.exports.prototype = {
-
     configure: function(requireCapitalizedConstructors) {
         assert(
-            typeof requireCapitalizedConstructors === 'boolean',
-            'requireCapitalizedConstructors option requires boolean value'
-        );
-        assert(
-            requireCapitalizedConstructors === true,
-            'requireCapitalizedConstructors option requires true value or should be removed'
+            requireCapitalizedConstructors === true || Array.isArray(requireCapitalizedConstructors.allExcept),
+            'requireCapitalizedConstructors option requires object of exceptions or true value'
         );
+        this._allowedConstructors = {};
+
+        var allExcept = requireCapitalizedConstructors.allExcept;
```

Comment:
```1.foobar``` throws, because JavaScript thinks that the ```.``` is the decimal point and expects more numbers after it (and ```foobar``` is not a number). So you could say ```1.foobar``` is an illegal numerical expression, which therefore throws.
If you add a second point like ```1..foobar``` the second point ends the numerical expression and acts as "property access point" (is there an official word for this?). And since there is no ```.foobar``` on numbers it returns undefined.